pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -422,15 +422,77 @@ func setNodeStatusFn(status *operatorv1.NodeStatus) v1helpers.UpdateStaticPodSta
 	}
 }
 
+type revisionDescriptionPrinter struct {
+	nodes          []operatorv1.NodeStatus
+	latestRevision int32
+	progressing    bool
+}
+
+func newRevisionDescriptionPrinter(nodes []operatorv1.NodeStatus, latestRevision int32, progressing bool) *revisionDescriptionPrinter {
+	return &revisionDescriptionPrinter{nodes: nodes, latestRevision: latestRevision, progressing: progressing}
+}
+
+func pluralizedNode(i int) string {
+	if i < 2 {
+		return "node"
+	}
+	return "nodes"
+}
+
+func (r revisionDescriptionPrinter) String() string {
+	currentRevisions := sets.NewInt32()
+	latestCount := 0
+	availableCount := 0
+	for i := range r.nodes {
+		c := r.nodes[i].CurrentRevision
+		currentRevisions.Insert(c)
+		if c == r.latestRevision {
+			latestCount++
+		}
+		if c != 0 {
+			availableCount++
+		}
+	}
+
+	notAvailable := fmt.Sprintf("%d %s are not available and ", availableCount, pluralizedNode(availableCount))
+	if availableCount == len(r.nodes) {
+		notAvailable = ""
+	}
+	if availableCount == 0 {
+		return "none of the nodes are available"
+	}
+
+	if !currentRevisions.Has(r.latestRevision) {
+		if r.progressing {
+			return fmt.Sprintf("%snodes progressing towards revision %d", notAvailable, r.latestRevision)
+		}
+		return fmt.Sprintf("%snone of the nodes reached latest available revision %d", notAvailable, r.latestRevision)
+	}
+	if currentRevisions.Equal(sets.NewInt32(r.latestRevision)) {
+		return fmt.Sprintf("%sall nodes reached revision %d", notAvailable, r.latestRevision)
+	}
+	if currentRevisions.Has(r.latestRevision) {
+		oldRevisionStringList := []string{}
+		for _, i := range currentRevisions.Difference(sets.NewInt32(r.latestRevision)).List() {
```

Comment:
loop over currentRevisions and exclude `latestRevision`. No need for set arithmetic.